idaes/core/control_volume1d.py

```diff
@@ -426,19 +426,15 @@ def _add_material_balance_common(self,
         # Check if reaction package exists, and get units
         if hasattr(self.config, "reaction_package"):
             if self.config.reaction_package is not None:
-                rxn_units = self.config.reaction_package.get_metadata().get_derived_units
-                if rxn_units('length') is not None:
-                    if (self.reactions[self.flowsheet().time.first(),
-                                       self.length_domain.first()]
-                            .get_reaction_rate_basis() == MaterialFlowBasis.molar):
-                        rxn_flow_l_units = units('flow_mole')/units('length')
-                    elif (self.reactions[self.flowsheet().time.first(),
-                                         self.length_domain.first()]
-                          .get_reaction_rate_basis() == MaterialFlowBasis.mass):
-                        rxn_flow_l_units = units('flow_mass')/units('length')
-                    else:  # reaction basis not defined
-                        rxn_flow_l_units = None
-                else:  # reaction package has no unit container objects
+                if (self.reactions[self.flowsheet().time.first(),
+                                   self.length_domain.first()]
+                        .get_reaction_rate_basis() == MaterialFlowBasis.molar):
+                    rxn_flow_l_units = units('flow_mole')/units('length')
+                elif (self.reactions[self.flowsheet().time.first(),
+                                     self.length_domain.first()]
+                      .get_reaction_rate_basis() == MaterialFlowBasis.mass):
+                    rxn_flow_l_units = units('flow_mass')/units('length')
+                else:  # reaction basis not defined
```

Comment:
Minor comment, this case which should raise an `Exception` of some type, as if the basis is "other" then we don't know what to do and should not try guessing (and `None` is not the right answer).

This test might already exist somewhere else, but it is probably worth adding here anyway - I would probably go with a `PropertyPackageError`.

Reply:
Regarding your second point, what is your opinion of adding a similar `Exception` to the property package "other basis" case on line 400?

Reply:
It should not be in the property package itself (otherwise there would be no point to having the option) - that option is there to allow for custom cases, but precludes the use of control volumes (hence the check should be in the control volumes).

Reply:
To clarify, I'm asking if we should add a `PropertyPackageError` to the control volume if the flow basis is not found to be Molar or Mass.

Reply:
Now that I look at that, I am not sure - we are basically saying we will accept "other" as a basis and assume you know what you are doing. So, we should do the same here then, and not raise the exception, just set units to `None`.